addons/hr_contract/models/resource.py

```diff
@@ -28,17 +30,34 @@ def transfer_leaves_to(self, other_calendar, resources=None, from_date=None):
             'calendar_id': other_calendar.id,
         })
 
+    @api.depends('contract_ids', 'contract_ids.state')
     def _compute_contracts_count(self):
-        count_data = self.env['hr.contract']._read_group(
-            [('resource_calendar_id', 'in', self.ids)],
-            ['resource_calendar_id'],
-            ['__count'])
-        mapped_counts = {resource_calendar.id: count for resource_calendar, count in count_data}
+        contracts_data = self.env['hr.contract']._read_group(
+            [('resource_calendar_id', 'in', self.ids), ('company_id', 'in', self.env.companies.ids)],
+            ['resource_calendar_id', 'state'],
+            ['__count']
```

Comment:
```suggestion
            domain=[('resource_calendar_id', 'in', self.ids), ('company_id', 'in', self.env.companies.ids)],
            fields=['resource_calendar_id', 'state'],
            aggregates=['__count']
```
I prefer for visibility; it's easier to read; what do you think ? :)